Assets/MirageXR/Player/Scripts/Spatial/InfoScreenSpatialView.cs

```diff
@@ -43,9 +62,319 @@ public void Initialize(ActivityStep step)
             UpdateView(step); 
         }
         
+        private void Update()
+        {
+            if (Input.GetMouseButtonDown(0)) // left mouse button
+            {
+                HandleClick();
+            }
+            
+            foreach (var linkId in _splineInstances.Keys)
+            {
+                if (!_hyperlinkInstances.TryGetValue(linkId, out var hyperlinkInstance) || hyperlinkInstance == null)
+                {
+                    CleanupLink(linkId); 
+                    continue;
+                }
+                var currentStartPosition = GetLinkWorldPosition(linkId);
+                var currentEndPosition = hyperlinkInstance.transform.position;
+                
+                if (_previousPositions.TryGetValue(linkId, out var previousPositions) &&
+                    previousPositions.startPosition == currentStartPosition &&
+                    previousPositions.endPosition == currentEndPosition)
+                {
+                    continue;
+                }
+
+                UpdateSplineLine(linkId);
+                _previousPositions[linkId] = (currentStartPosition, currentEndPosition);
+            }
+        }
+        private void CleanupLink(string linkId)
+        {
+            _hyperlinkInstances.Remove(linkId);
+            if (_splineInstances.TryGetValue(linkId, out var spline))
+            {
+                Destroy(spline.gameObject);
+                _splineInstances.Remove(linkId);
+            }
+            _previousPositions.Remove(linkId);
+        }
+        
+        private void HandleClick()
+        {
+            var mousePosition = Input.mousePosition;
```

Comment:
This will not work on the smart glasses - needs to be a Spatial Pointer Device interactionPosition or some such, see here: https://docs.unity3d.com/Packages/com.unity.polyspatial.visionos@1.3/manual/PolySpatialInput.html

Reply:
There are good examples on the page. If not via direct scripting (which I think is better), then HandClick() could also be triggered via the XRTouchSpaceInteractor, but I think this is more complicated, and @cawke is currently updating the use of XRTouchSpaceInteractor?